lib/money/money.rb

```diff
@@ -619,6 +635,13 @@ def decimal_mark
 
   private
 
+  def dup_with(options = {})
+    fractional ||= options[:fractional] || self.fractional
+    currency ||= options[:currency] || self.currency
+    bank ||= options[:bank] || self.bank
+    self.class.new(fractional, currency, bank)
```

Comment:
Probably need to use `bank: bank` as the last argument here given the new changes